src/jit/lsra.cpp

```diff
@@ -6373,6 +6387,24 @@ void LinearScan::unassignPhysReg(RegRecord* regRec, RefPosition* spillRefPositio
     {
         regRec->assignedInterval = regRec->previousInterval;
         regRec->previousInterval = nullptr;
+#ifdef _TARGET_ARM_
+        // Update overlapping floating point register for TYP_DOUBLE
+        if (regRec->assignedInterval->registerType == TYP_DOUBLE)
+        {
+            assert(isFloatRegType(regRec->registerType));
+
+            regNumber  nextRegNum = REG_NEXT(regRec->regNum);
+            RegRecord* nextRegRec = getRegisterRecord(nextRegNum);
+
+            nextRegRec->assignedInterval = nextRegRec->previousInterval;
+            nextRegRec->previousInterval = nullptr;
+
+            // TODO-ARM-Throughput: For ARM, we can reuse regRec->previousInterval,
+            // if following assertion is always true.
+            // Please take a look at tryAllocateFreeReg().
+            assert(nextRegRec->assignedInterval == regRec->assignedInterval);
```

Comment:
What about the case where interval0 in d0(f0/f1) becomes inactive, and f0 is assigned to a `TYP_FLOAT` (say interval1). At this point both `previousInterval` fields reference interval0. Then interval1 becomes inactive and is reallocated to yet another `TYP_FLOAT`. At that point, its `previousInterval` will be set to interval1, but the `previousInterval` of f1 is still interval0. Have I missed something that will handle this case?

---
In reply to: [112926942](https://github.com/dotnet/coreclr/pull/10972#discussion_r112926942) [](ancestors = 112926942)